lib/ProductOpener/Images.pm

```diff
@@ -1181,6 +1181,13 @@ sub process_image_move ($user_id, $code, $imgids, $move_to, $ownerid) {
 						result => $ok
 					}
 				);
+
+            # Unselect images with matching imgid
+        	foreach my $selected_imgid (keys %{$product_ref->{images}}) {
+          		if ($selected_imgid eq $imgid) {
```

Comment:
Hi @himanshi-154 , what we want is to unselect entries that have the "imgid" subfield equal, to $imgid. Here you are unselecting the source image, not the selected images.

I added some steps in the issue description so that you can verify if the issue is solved: https://github.com/openfoodfacts/openfoodfacts-server/issues/9793#issuecomment-1994223630